webapp/src/ts/modules/reports/reports-more-menu.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, EventEmitter, OnDestroy, OnInit, Output } from '@angular/core';
+import { combineLatest, Subscription } from 'rxjs';
+import { Store } from '@ngrx/store';
+
+import { Selectors } from '@mm-selectors/index';
+import { AuthService } from '@mm-services/auth.service';
+import { GlobalActions } from '@mm-actions/global';
+import { ResponsiveService } from '@mm-services/responsive.service';
+import { SessionService } from '@mm-services/session.service';
+import { OLD_ACTION_BAR_PERMISSION } from '@mm-components/actionbar/actionbar.component';
+
+@Component({
+  selector: 'mm-reports-more-menu',
+  templateUrl: './reports-more-menu.component.html'
+})
+export class ReportsMoreMenuComponent implements OnInit, OnDestroy {
+  @Output() exportReports: EventEmitter<any> = new EventEmitter();
+
+  private globalActions: GlobalActions;
+  private hasExportPermission = false;
+  private hasEditPermission = false;
+  private hasDeletePermission = false;
+  private hasUpdatePermission = false;
+  private selectMode = false;
+  private loadingContent;
+  private snapshotData;
+  private isOnlineOnly;
```

Comment:
Same here with `import type { Data } from '@angular/router';` at the top

```suggestion
  private loadingContent: boolean;
  private snapshotData: Data;
  private isOnlineOnly: boolean;
```

Reply:
Nice!